content/en/blog/_posts/2022-04-19-non-graceful-node-shutdown.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: blog
+title: "Kubernetes 1.24: Introducing Non-Graceful Node Shutdown Alpha"
+date: 2022-04-19T10:00:00-08:00
+slug: kubernetes-1-24-non-graceful-node-shutdown-alpha
+---
+
+**Authors** Xing Yang and Yassine Tijani (VMware)
+
+Kubernetes v1.24 introduces alpha support for [Non-Graceful Node Shutdown](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2268-non-graceful-shutdown). This feature allows stateful workloads to failover to a different node after the original node is shutdown or in a non-recoverable state such as the hardware failure or broken OS.
+
+## How is this different from Graceful Node Shutdown
+
+You might have heard about the [Graceful Node Shutdown](https://kubernetes.io/blog/2021/04/21/graceful-node-shutdown-beta/) feature and are wondering how the Non-Graceful Node Shutdown feature is different from that. Graceful Node Shutdown introduced a way to detect a node shutdown and handle it gracefully. A Node Shutdown can be `Graceful` only if the node shutdown action can be detected by Kubelet's Node Shutdown Manager. However, there are cases where a node shutdown action may not be detected by Kubelet's Node Shutdown Manager. This could happen either because the shutdown command does not trigger the inhibitor locks mechanism used by Kubelet or because of a user error, i.e., the `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are not configured properly.
```

Comment:
How about this:
```markdown
You might have heard about the [Graceful Node Shutdown](/docs/concepts/architecture/nodes/#graceful-node-shutdown) capability of Kubernetes,
and are wondering how the Non-Graceful Node Shutdown feature is different from that. Graceful Node Shutdown
allows Kubernetes to detect when a node is shutting down cleanly, and handle that situation appropriately.
A Node Shutdown can be “graceful” only if the node shutdown action can be detected by the kubelet ahead
of the actual shutdown. However, there are cases where a node shutdown action may not be detected by
the kubelet. This could happen either because the shutdown command does not trigger the systemd inhibitor
locks mechanism that kubelet relies up, because of a system fault, or because of a configuration error
(the `ShutdownGracePeriod` and `ShutdownGracePeriodCriticalPods` are not configured properly).

Graceful node shutdown relies on Linux-specific support. The kubelet does not watch for upcoming
shutdowns on Windows nodes (this may change in a future Kubernetes release).
```

I wrapped it. Articles are either to review and to localize if you write the text with line breaks.

Reply:
Updated with your suggestion. The only thing I didn't take is this line "because of a system fault" because this is not necessarily true. When I tested Graceful Node Shutdown on my setup, only one of the three shutdown commands listed here (https://kubernetes.io/blog/2021/04/21/graceful-node-shutdown-beta/#how-does-it-work) worked. When I shut down a VM from the vSphere UI, it does not trigger the inhibitor locks either.